### ISSUE TITLE
oCIS fails to start with `MICRO_REGISTRY` set to etcd

### ISSUE THREAD

**Issue by dragotin** [owncloud/ocis#3640]
As already discussed in the chat, my local build of ocis (recent versions) fails to start with the following error message:
```
2022-05-01T13:40:49+02:00 INF running on 6 cpus service=ocis
{"level":"fatal","time":"2022-05-01T13:40:49+02:00","message":"Error configuring broker: cannot init while connected"}
Process 10537 has exited with status 1
```
The http broker of go-micro that we use by default checks in the Init method if the broker is already running: https://github.com/asim/go-micro/blob/master/broker/http.go#L462

While this patch:
```
diff --git a/broker/http.go b/broker/http.go
index 64388f24..07018a7f 100644
--- a/broker/http.go
+++ b/broker/http.go
@@ -5,7 +5,6 @@ import (
        "bytes"
        "context"
        "crypto/tls"
-       "errors"
        "fmt"
        "io"
        "math/rand"
@@ -461,7 +460,7 @@ func (h *httpBroker) Init(opts ...Option) error {
        h.RLock()
        if h.running {
                h.RUnlock()
-               return errors.New("cannot init while connected")
+               return nil // errors.New("cannot init while connected")
        }
        h.RUnlock()
```
fixes the problem, this has a **smell**. 

Some questions that need to be clarified:
- How many brokers should be started when running in a single binary?
- As I understand the code, I guess it should be only one. In that case, I don't think it makes sense to call the `Init` function of the Broker again for each service using the broker.
-  What consequences does my proposed patch have?
- Why do only I see that problem on my machine :fearful:, it seems to work for anyone else? 

I would feel better if I understood this problem ;-)

@refs let me reference you here as a go micro maintainer.

**Comment by kobergj** [comment]
@dragotin I could reproduce by setting one of the envvars `"MICRO_REGISTRY"` or `"MICRO_BROKER_ADDRESS"`. 
Needs further debugging why it fails locally on your machine.

**Comment by dragotin** [comment]
Interesting. Yes, I have indeed the envvar `MICRO_REGISTRY` set to `etcd` 

That means using etcd is not longer needed/supported?

**Comment by micbar** [comment]
Related to #3829

**Comment by micbar** [comment]
Fixed by #4576
